cas-client-core/src/main/java/org/jasig/cas/client/validation/Cas20ServiceTicketValidator.java

```diff
@@ -113,8 +110,45 @@ protected Assertion parseResponseFromServer(final String response) throws Ticket
         return assertion;
     }
 
-    protected String parseProxyGrantingTicketFromResponse(final String response) {
-        return XmlUtils.getTextForElement(response, "proxyGrantingTicket");
+    protected String retrieveProxyGrantingTicket(final String response) {
+        final List<String> values = XmlUtils.getTextForElements(response, "proxyGrantingTicket");
+        for (final String value : values) {
+            if (value != null) {
+                /*if (!value.startsWith("PGTIOU-")) {
+                    return retrieveProxyGrantingTicketViaEncryption(value);
+                }*/
+                if (value.startsWith("PGTIOU-")) {
```

Comment:
The value of the PGT never starts out with PGTIOU. That's a different attribute. You want to check for `PGT-` instead.

Reply:
Yes, sure. Two kind of data can be retrieved in the `proxyGrantingTicket` attribute:  either a PGTIOU which starts with "PGTIOU-"  or an encrypted PGT (which is base64 encoded as well).

A cleaned example:

```xml
<cas:serviceResponse xmlns:cas='http://www.yale.edu/tp/cas'>
    <cas:authenticationSuccess>
        <cas:user>jleleu</cas:user>
        <cas:proxyGrantingTicket>PGTIOU-5-Z3WQsrcCHD6zbcLLugje8Bup8mOIp9itwWLsZT8oj4D4N4-3IFvTzxMzDgRhES-zqigcastest</cas:proxyGrantingTicket>
        <cas:attributes>

...
            <cas:proxyGrantingTicket>PKbONdwOCU3EYBmgethsNF3XxLbxxmlEtlMPka4Fmy6Gay1w4p28SXs2VXgle90na0/DgtoMGSO0WbXS96T4/FVk4x19Ex9KteA46WJv42dYBeWlfzPT4hI91avNImJiqQhATH7EOe63jSZCHKlZQH4eyYlyVwpjAtatnLpLMD4=</cas:proxyGrantingTicket>

...

            </cas:attributes>
    </cas:authenticationSuccess>
</cas:serviceResponse>
```

So I want to know if it's a PGTIOU or an encrypted PGT. Hence the test.